xdsl/transforms/convert_memref_to_ptr.py

```diff
@@ -19,6 +19,39 @@
 from xdsl.utils.exceptions import DiagnosticException
 from xdsl.utils.hints import isa
 
+_index_type = builtin.IndexType()
+
+
+def get_bytes_offset(
+    elements_offset: SSAValue, element_type: Attribute, builder: Builder
+) -> SSAValue:
+    """
+    Returns the offset in bytes given an offset in elements and the element type.
+    """
+    bytes_per_element_op = builder.insert_op(
+        ptr.TypeOffsetOp(element_type, _index_type)
+    )
+    bytes_offset = builder.insert_op(
+        arith.MuliOp(elements_offset, bytes_per_element_op)
+    )
+    bytes_per_element_op.offset.name_hint = "bytes_per_element"
+    bytes_offset.result.name_hint = "scaled_pointer_offset"
+
+    return bytes_offset.result
+
+
+def get_offset_pointer(
+    pointer: SSAValue,
+    bytes_offset: SSAValue,
+    builder: Builder,
+) -> SSAValue:
+    """
+    Returns the pointer scaled by an offset in bytes.
```

Comment:
Is scaled the right term? Maybe 
```suggestion
    Returns the provided pointer offset by a given number of bytes.
```